packages/replayio/src/utils/installation/getLatestReleases.ts

```diff
@@ -5,24 +5,24 @@ import { runtimeMetadata } from "./config";
 import { debug } from "./debug";
 import { Release } from "./types";
 
-const { platform, runtime } = runtimeMetadata;
+const { platform, runtime, architecture } = runtimeMetadata;
 
 export async function getLatestRelease() {
   debug("Fetching release metadata");
 
   const response = await fetch(`${replayAppHost}/api/releases`);
   const json = (await response.json()) as Release[];
-  const latestRelease = json.find(release => {
-    if (release.platform === platform && release.runtime === runtime) {
-      if (platform === "macOS") {
-        return process.arch.startsWith("arm") === release.releaseFile.includes("arm");
-      } else {
-        return true;
-      }
-    }
-  });
+  const latestRelease = json.find(
+    release =>
+      release.platform === platform &&
+      release.runtime === runtime &&
+      (release.architecture === architecture || release.architecture === null)
```

Comment:
That being said, this change would temporarily break Mac clients, no?

Right now the only release for 20240508 is an x86 release and that release has `"architecture": null`

I think this change would make a Mac with silicon incorrectly match that release.

Reply:
Agreed, but we did just do a release that now includes entries for both arm and x86 on mac:

```json
[
  {
    "buildId": "linux-chromium-20240510-6bb238250bf2-99bab3984009",
    "runtime": "chromium",
    "version": "108.0.0",
    "platform": "linux",
    "architecture": "x86_64",
    "time": "2024-05-10T21:50:51.972Z",
    "buildFile": "linux-chromium-20240510-6bb238250bf2-99bab3984009.tar.xz",
    "releaseFile": "linux-replay-chromium.tar.xz"
  },
  {
    "buildId": "macOS-chromium-20240510-6bb238250bf2-99bab3984009",
    "runtime": "chromium",
    "version": "108.0.0",
    "platform": "macOS",
    "architecture": "x86_64",
    "time": "2024-05-10T21:47:49.956Z",
    "buildFile": "macOS-chromium-20240510-6bb238250bf2-99bab3984009.dmg",
    "releaseFile": "Replay-Chromium.dmg"
  },
  {
    "buildId": "macOS-chromium-20240510-6bb238250bf2-99bab3984009",
    "runtime": "chromium",
    "version": "108.0.0",
    "platform": "macOS",
    "architecture": "arm",
    "time": "2024-05-10T21:47:19.557Z",
    "buildFile": "macOS-chromium-20240510-6bb238250bf2-99bab3984009-arm.dmg",
    "releaseFile": "Replay-Chromium-arm.dmg"
  },
  // ...
]
```

I think that resolves this issue?

Reply:
Right! At the time I left this comment that wasn't the case, and our release process is somewhat broken (particularly with respect to Mac) where it wasn't clear that the next release would address both arm and x86 platforms, so this could have broken things.

Looking at the current releases (from yesterday) we have x86 for MacOS and Linux and arm for Mac.

So yeah, I think this should be okay now. I think we can address your TODO comment and remove the `null` case now too since it looks like the API now sends "unknown" for the older releases?